### PR TITLE
[KYUUBI #5918] Kyuubi BeeLine should check the relocated TTransportException

### LINKED ISSUE #5918: [Bug] Kyuubi BeeLine should check the relocated TTransportException
### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### Search before asking

- [X] I have searched in the [issues](https://github.com/apache/kyuubi/issues?q=is%3Aissue) and found no similar issues.

### Describe the bug

When Kyuubi beeline executes incorrect SQL,it will exit with exception:
java.lang.NoClassDefFoundError: org/apache/thrift/transport/TTransportException

### Affects Version(s)

1.8.0

### Are you willing to submit PR?

- [X] Yes. I would be willing to submit a PR with guidance from the Kyuubi community to fix.
- [ ] No. I cannot submit a PR at this time.

### PR DESCRIPTION
**Pull request by hadoopkandy**
# :mag: Description
## Issue References 🔗

This pull request fixes #5918, when Kyuubi beeline executes incorrect SQL,it will exit with exception:
```
java.lang.NoClassDefFoundError: org/apache/thrift/transport/TTransportException
```

## Describe Your Solution 🔧
1.Supply `org.apache.thrift:libthrift` to `kyuubi-beeline` module.
2.We should override BeeLine's handleSQLException method,makes it compatible
  As we discussed, only `org.apache.kyuubi.shaded.thrift.transport.TTransportException` will be received by `kyuubi-beeline`,so it‘s  unnecessary  to add libthrift dependency to `kyuubi-beeline`.

## Types of changes :bookmark:

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

#### Related Unit Tests

---

# Checklists
## 📝 Author Self Checklist

- [ ] My code follows the [style guidelines](https://kyuubi.readthedocs.io/en/master/contributing/code/style.html) of this project
- [ ] I have performed a self-review
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] This patch was not authored or co-authored using [Generative Tooling](https://www.apache.org/legal/generative-tooling.html)

## 📝 Committer Pre-Merge Checklist

- [x] Pull request title is okay.
- [x] No license issues.
- [x] Milestone correctly set?
- [ ] Test coverage is ok
- [x] Assignees are selected.
- [x] Minimum number of approvals
- [x] No changes are requested

**Be nice. Be informative.**

### CODE REVIEW

**Comment by pan3793** [comment]
This should be caused by https://github.com/apache/kyuubi/pull/5783.

The current way of extending beeline is fragile, it's really hard to maintain the transitive deps, I suppose we need to rework the kyuubi-beeline module as we did in kyuubi-hive-jdbc and get rid of hive deps eventually.

**Review by pan3793 — APPROVED** [review]

File: kyuubi-hive-beeline/pom.xml [hunk 1]
```diff
@@ -46,6 +46,11 @@
             <version>${project.version}</version>
         </dependency>
 
+        <dependency>
+            <groupId>org.apache.thrift</groupId>
+            <artifactId>libthrift</artifactId>
```
Comment (cxzl25) [inline kyuubi-hive-beeline/pom.xml:51]: Beeline currently does not have a shade, and the code also has some thrift-related classes. 
If the server returns the thrift exception class of the shade, beeline may not be able to handle it correctly.

org.apache.hive.beeline.BeeLine#handleSQLException

https://github.com/apache/hive/blob/81b394104b08efdc42d782d444462f4fdc12a8fd/beeline/src/java/org/apache/hive/beeline/BeeLine.java#L2179-L2197

Comment (pan3793) [reply kyuubi-hive-beeline/pom.xml:51]: yea, strictly speaking, we must fork and get rid of vanilla beeline to fix it thoroughly

Comment (cxzl25) [reply kyuubi-hive-beeline/pom.xml:51]: It sounds like we have to be compatible with two thrift exceptions, one from hiveserver2 and one from kyuubi server.

**Review by cxzl25 — APPROVED** [review]

File: kyuubi-hive-beeline/pom.xml (hunk 1 above)
Comment (pan3793) [reply kyuubi-hive-beeline/pom.xml:51]: let's try this way, the method visibility is package level, we can overwrite it in sub-class that located at the same pacakge

**Review by pan3793 — APPROVED** [review]

**Comment by pan3793** [comment]
I think the current fix is valid, @cxzl25 do you have any more concerns?

File: kyuubi-hive-beeline/src/main/java/org/apache/hive/beeline/KyuubiBeeLine.java [hunk 2]
```diff
@@ -299,4 +302,56 @@ int runInit() {
...
+          error(loc("hs2-could-not-open-connection"));
+          break;
+        case TTransportException.TIMED_OUT:
+          error(loc("hs2-connection-timed-out"));
+          break;
+        case TTransportException.UNKNOWN:
+          error(loc("hs2-unknown-connection-problem"));
+          break;
+        default:
+          error(loc("hs2-unexpected-error"));
```
Comment (cxzl25) [inline kyuubi-hive-beeline/src/main/java/org/apache/hive/beeline/KyuubiBeeLine.java:330]: If it is Hiveserver2 connected by kyuubi beeline, the exception returned by Hiveserver2 is `org.apache.thrift.transport.TTransportException`. 

It should not be matched here?

cc @zhouyifan279

Comment (cxzl25) [reply kyuubi-hive-beeline/src/main/java/org/apache/hive/beeline/KyuubiBeeLine.java:330]: My description here is wrong, please ignore it.
Thank you for your contribution!
LGTM

**Comment by pan3793** [comment]
Thanks, merged to master

**Pull request closed without merging**